spec/models/email_settings_spec.rb

```diff
@@ -0,0 +1,102 @@
+      'cfp_start_date' => 'Unknown',
+      'venue' => 'Unknown',
+      'venue_address' => 'Unknown'
+    }
+  end
+
+  it 'has a valid factory' do
+    expect(build(:email_settings)).to be_valid
+  end
+
+  describe '#get_values' do
+    it 'returns correct key-value pairs' do
+      expect(conference.email_settings.get_values(conference, user)).to eq expected_hash
+    end
+
+    context 'conference has cfp' do
+      before do
+        conference.program.update_attributes(cfp: create(:cfp,
+                                                         start_date: Date.new(2014, 04, 29),
+                                                         end_date: Date.new(2014, 05, 06)))
+      end
+
+      it 'returns hash with cfp start and end date' do
+        cfp_dates_hash = { 'cfp_start_date' => Date.new(2014, 04, 29), 'cfp_end_date' => Date.new(2014, 05, 06) }
+        expected_hash.merge!(cfp_dates_hash)
+        expect(conference.email_settings.get_values(conference, user)).to eq expected_hash
+      end
+    end
+
+    context 'conference has venue' do
+      before { conference.update_attributes(venue: create(:venue)) }
+
+      it 'returns hash with venue and venue_address' do
+        venue_hash = { 'venue' => 'Suse Office', 'venue_address' => 'Maxfeldstrasse 5, Nuremberg, Germany' }
+        expected_hash.merge!(venue_hash)
+        expect(conference.email_settings.get_values(conference, user)).to eq expected_hash
+      end
+    end
+
+    context 'conference has registration period' do
+      before do
+        conference.update_attributes(registration_period: create(:registration_period,
+                                                                 start_date: Date.new(2014, 05, 03),
+                                                                 end_date: Date.new(2014, 05, 05)))
+      end
+
+      it 'returns hash with registration_start_date and registration_end_date' do
+        registration_period_hash = { 'registration_start_date' => Date.new(2014, 05, 03), 'registration_end_date' => Date.new(2014, 05, 05) }
+        expected_hash.merge!(registration_period_hash)
+        expect(conference.email_settings.get_values(conference, user)).to eq expected_hash
+      end
+    end
+
+    context 'conference has event' do
+      it 'returns hash with eventtitle and proposalslink' do
+        event_hash = { 'eventtitle' => 'Talk about talks', 'proposalslink' => 'http://localhost:3000/conference/goto/program/proposal' }
+        expected_hash.merge!(event_hash)
+        expect(conference.email_settings.get_values(conference, user, event)).to eq expected_hash
+      end
+    end
+  end
+
+  describe '#generate_event_mail' do
+    let(:event_template) do
+      "Dear {name}\n\nWe are very pleased" \
+      'to inform you that your submission {eventtitle} has been accepted for the conference {conference}.'
+    end
+
+    it 'replaces fillers in template' do
+      expected_text = "Dear John Doe\n\nWe are very pleased" \
```

Comment:
@ChrisBr  Thanks for bringing it up. I have added this test [here](https://github.com/openSUSE/osem/pull/917/files#diff-4dfcd74d95dd9208044a4c1f32b3f24bR36). `get_values` is responsible for setting the hash, `generate_event_mail` merely uses the value set by `get_values`. Do you think I should test it in `generate_event_mail` too?